tools/topology/sof-apl-pcm512x.m4

```diff
@@ -150,9 +185,21 @@ DAI_CONFIG(SSP, 5, 0, SSP5-Codec,
 		SSP_TDM(2, 32, 3, 3),
 		SSP_CONFIG_DATA(SSP, 5, 24)))
 
-# 3 HDMI/DP outputs (ID: 1,2,3)
-DAI_CONFIG(HDA, 0, 1, iDisp1)
-DAI_CONFIG(HDA, 1, 2, iDisp2)
-DAI_CONFIG(HDA, 2, 3, iDisp3)
+# DMIC (ID: 1)
+DAI_CONFIG(DMIC, 0, 1, dmic01,
+	   DMIC_CONFIG(1, 500000, 4800000, 40, 60, 48000,
+		DMIC_WORD_LENGTH(s32le), 400, DMIC, 0,
+		PDM_CONFIG(DMIC, 0, FOUR_CH_PDM0_PDM1)))
+
+# DMIC16kHz (ID: 2)
+DAI_CONFIG(DMIC, 1, 2, dmic16k,
+	   DMIC_CONFIG(1, 500000, 4800000, 40, 60, 16000,
+		DMIC_WORD_LENGTH(s16le), 400, DMIC, 1,
+		PDM_CONFIG(DMIC, 1, STEREO_PDM0)))
+
+# 3 HDMI/DP outputs (ID: 3,4,5)
+DAI_CONFIG(HDA, 0, 3, iDisp1)
+DAI_CONFIG(HDA, 1, 4, iDisp2)
+DAI_CONFIG(HDA, 2, 5, iDisp3)
```

Comment:
Renumbering the iDisp ids is a bit nasty as this will create 1:1 dependency between this and the kernel machine driver patch. But ok, this is not in upstream at all, so ok.

Reply:
@kv2019i Agree: DAI IDs are used for matching, so they are a part of the ABI. As such they should use a more robust assignment method than just a loop index... We could add some mapping in follow-up patches on the driver side.